recipes/physx/4.1.1/conanfile.py

```diff
@@ -0,0 +1,264 @@
+import os
+import time
+
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+
+class PhysXConan(ConanFile):
+    name = "physx"
+    description = "The NVIDIA PhysX SDK is a scalable multi-platform " \
+                  "physics solution supporting a wide range of devices, " \
+                  "from smartphones to high-end multicore CPUs and GPUs."
+    license = "BSD-3-Clause"
+    topics = ("conan", "PhysX", "physics")
+    homepage = "https://github.com/NVIDIAGameWorks/PhysX"
+    url = "https://github.com/conan-io/conan-center-index"
+    exports_sources = ["CMakeLists.txt", "patches/**"]
+    generators = "cmake"
+    settings = "os", "compiler", "arch", "build_type"
+    short_paths = True
+    options = {
+        "shared": [True, False],
+        "release_build_type": ["profile", "release"],
+        "enable_simd": [True, False],
+        "enable_float_point_precise_math": [True, False],
+        "fPIC": [True, False],
+    }
+    default_options = {
+        "shared": True,
+        "release_build_type": "release",
+        "enable_simd": True,
+        "enable_float_point_precise_math": False,
+        "fPIC": True,
+    }
+
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def config_options(self):
+        del self.options.fPIC # fpic is managed by physx build process (in physx/source/compiler/cmake/CMakeLists.txt)
+        if self.settings.build_type != "Release":
+            del self.options.release_build_type
+        if self.settings.os != "Windows":
+            del self.options.enable_float_point_precise_math
+        if self.settings.os not in ["Windows", "Android"]:
+            del self.options.enable_simd
+
+    def configure(self):
+        os = str(self.settings.os)
```

Comment:
stringifying `os`, `build_type`, `compiler`, `runtime` is not absolutely necessary.